Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptorTests.swift

```diff
@@ -128,11 +130,22 @@ class URLSessionInterceptorTests: XCTestCase {
         )
         Global.sharedTracer = Tracer.mockAny()
         defer { Global.sharedTracer = DDNoopGlobals.tracer }
+        let sessionWithCustomFirstPartyHosts = URLSession(
+            configuration: .default,
+            delegate: DDURLSessionDelegate(
+                firstPartyHosts: [alternativeFirstPartyRequest.url!.host!]
+            ),
+            delegateQueue: nil
+        )
```

Comment:
If we need to mock only part of the `URLSession` interface, it would be more readable to create partial mock:
```swift
let sessionWithCustomFirstPartyHosts: URLSession = .mockWith(
   delegate: DDURLSessionDelegate(firstPartyHosts: [alternativeFirstPartyRequest.url!.host!])
)
```
as we do for many other things in there.